redbeams/src/test/java/com/sequenceiq/redbeams/service/network/AwsSubnetValidatorTest.java

```diff
@@ -0,0 +1,40 @@
+package com.sequenceiq.redbeams.service.network;
+
+
+import java.util.List;
+
+import org.junit.Test;
+
+import com.sequenceiq.cloudbreak.cloud.model.CloudSubnet;
+import com.sequenceiq.redbeams.exception.BadRequestException;
+
+public class AwsSubnetValidatorTest {
+
+    private AwsSubnetValidator underTest = new AwsSubnetValidator();
+
+    private CloudSubnet cloudSubnet1 = new CloudSubnet("subnet1", "", "AZ1", "");
+
+    private CloudSubnet cloudSubnet2 = new CloudSubnet("subnet2", "", "AZ1", "");
+
+    private CloudSubnet cloudSubnet3 = new CloudSubnet("subnet3", "", "AZ2", "");
+
+    @Test
+    public void testOk() {
+        underTest.validate(List.of(cloudSubnet1, cloudSubnet3), 2);
+    }
+
+    @Test
+    public void testOkWithMore() {
+        underTest.validate(List.of(cloudSubnet1, cloudSubnet2, cloudSubnet3), 2);
+    }
+
+    @Test(expected = BadRequestException.class)
+    public void testNOkWithTooFewSubnet() {
+        underTest.validate(List.of(cloudSubnet1), 2);
+    }
+
+    @Test(expected = BadRequestException.class)
+    public void testNOkWithTooFewAz() {
+        underTest.validate(List.of(cloudSubnet1, cloudSubnet2), 2);
```

Comment:
I think it shoulf be great to check the error message as well.

Reply:
fixed